### PR TITLE
Added crhook hook style

### PR DESCRIPTION
**Pull request by SpookyScaryDev**
Added crhook hookstyle as described by @inf1niti in #303 
At the moment, haste still affects throw speed when using crhook, as that was the original behaviour crhook was removed. I also changed the naming of the speed constants, since the speed for the new hook styles was called CR_THROW_SPEED.

### CODE REVIEW

File: src/grapple.c [hunk 1]
```diff
@@ -429,7 +430,10 @@ void GrappleThrow()
 	}
 
 	float hasteMultiplier =	(cvar("k_ctf_rune_power_hst") / 16) + 1;
-	float throwSpeed = cvar("k_ctf_hookstyle") != 3 ? CR_THROW_SPEED : THROW_SPEED;	
+	float throwSpeed = cvar("k_ctf_hookstyle") != 3 ? NEW_THROW_SPEED : THROW_SPEED;	
+	
+	if (cvar("k_ctf_hookstyle") == 4) throwSpeed = CR_THROW_SPEED;
+
```
Comment (inf1niti) [inline src/grapple.c:436]: might want to convert these to a series of `if` statements, keeping the ternary doesn't make a ton of sense anymore.

File: src/commands.c [hunk 2]
```diff
@@ -882,6 +884,7 @@ cmd_t cmds[] =
 	{ "hook_smooth", 				hooksmooth, 					0, 			CF_PLAYER | CF_MATCHLESS, 											CD_HOOKSMOOTH },
 	{ "hook_fast", 					hookfast, 					0, 			CF_PLAYER | CF_MATCHLESS, 											CD_HOOKFAST },
 	{ "hook_classic", 				hookclassic, 					0, 			CF_PLAYER | CF_MATCHLESS, 											CD_HOOKCLASSIC },
+	{ "crhook", 					hookcrhook, 					0, 			CF_PLAYER | CF_MATCHLESS, 											CD_HOOKCRHOOK },
```
Comment (inf1niti) [inline src/commands.c:887]: I know this was the original command for this hookstyle, but it makes it more difficult to find the various options if they're not following the same naming convention.  Lets stick with the `hook_` prefix for this as well.

**Review by inf1niti — CHANGES_REQUESTED** [review]
So I've tested this, and it all seems to work.  

If you guys have tested it as well, and are OK with the speed - I'm cool with merging this. 👍 

My only suggestion (besides the couple of inline comments left below) is that we no longer really need the `hook_fast` option - it was meant to be an option for people who didn't want to use the enhancements provided by the `hook_smooth` setting, but this hook is exactly the same (but faster).  

My suggestion would be to removing the existing `hook_fast` implementation entirely, and renaming `crhook` to `hook_fast`.  The `crhook` name honestly doesn't make sense, and this hook is faster than the existing fast hook anyways 😄 

That way, we can also keep the hook options to 3, which feels slightly more reasonable than 4.

**Comment by cbirkeland** [comment]
@inf1niti
Sounds good! Agree with naming. 
Thank you @inf1niti and @SpookyScaryDev .

**Comment by tcsabina** [comment]
@SpookyScaryDev,
let us know when ready to merge.

**Comment by tcsabina** [comment]
> @tcsabina Will do. Hope to do testing of the different options with the ctf community this week.

Hi @SpookyScaryDev,
Have you had a chance to test this? Are we ready to merge?

Let me know...

**Comment by tcsabina** [comment]
Guys,
Have you had some time to test this? Are we ready to merge?

Let me know...

**Comment by cbirkeland** [comment]
@SpookyScaryDev  is on vacation until 11th. Hence no reply :) From CTF Admins point of view, we are good to merge if crhook is back like it used to be before it got removed with the values we requested.
There was a night of testing but i think main focus was on the ctf bots that hiipe is working on.
Again, thanks!

**Comment by tcsabina** [comment]
Thanks @cbirkeland!
Let's wait a few more days...

**Comment by SpookyScaryDev** [comment]
If the code is ok, this should be ready to push now.

**Comment by tcsabina** [comment]
> If the code is ok, this should be ready to push now.

Thanks for the note. Merged...

**Pull request merged**